integration-libs/opf/cta/core/services/opf-dynamic-cta.service.ts

```diff
@@ -0,0 +1,252 @@
+
+      this.isCartPage && this.cartChangedListener();
+      !this.isCartPage && this.productChangedListener();
+
+      this.isOnsiteMessagingInit = true;
+    }
+  }
+
+  stopEvents() {
+    if (this.isOnsiteMessagingInit) {
+      this.subList.forEach((sub) => sub.unsubscribe());
+      this.subList = [];
+      this.globalFunctionsFacade.removeGlobalFunctions(
+        GlobalFunctionsDomain.GLOBAL
+      );
+      this.isOnsiteMessagingInit = false;
+    }
+  }
+
+  protected getNewScriptIdentifier(): string {
+    this.scriptIdentifiers.push(String(this.scriptIdentifiers.length + 1));
+    return String(
+      this.scriptIdentifiers[this.scriptIdentifiers.length - 1]
+    ).padStart(4, '0');
+  }
+
+  registerScriptReadyEvent() {
+    this.globalFunctionsFacade.registerGlobalFunctions({
+      paymentSessionId: '',
+      domain: GlobalFunctionsDomain.GLOBAL,
+    });
+  }
+
+  protected listenScriptReadyEvent() {
+    const sub = this.opfCtaFacade.listenScriptReadyEvent().subscribe({
+      next: (id: string) => {
+        this.isCartPage && this.dispatchCartEvents([id]);
+        !this.isCartPage && this.dispatchProductEvents([id]);
+      },
+    });
+    this.subList.push(sub);
+  }
+
+  protected dispatchCartEvents(scriptIdentifiers: Array<string>): void {
+    this.activeCartFacade
+      .takeActive()
+      .pipe(
+        take(1),
+        map((cart: Cart) => {
+          return { total: cart.totalPrice?.value };
+        }),
+        tap((cartTotalPrice) => {
+          const window = this.winRef.nativeWindow;
+          const dispatchEvent = window?.dispatchEvent;
+          if (dispatchEvent) {
+            dispatchEvent(
+              new CustomEvent(CtaEvent.OPF_CART_CHANGED, {
+                detail: { cart: cartTotalPrice, scriptIdentifiers },
+              })
+            );
+          }
+        })
+      )
+      .subscribe();
+  }
+
+  protected dispatchProductEvents(
+    scriptIdentifiers: Array<string>,
+    quantity = 1
+  ): void {
+    this.currentProductService
+      .getProduct()
+      .pipe(
+        take(1),
+        map((product: Product | null) => {
+          return [
+            {
+              price: {
+                sellingPrice: product?.price?.value,
+              },
+              quantity, // hard-coded until 'pdp counter change' event gets implemented
+            },
+          ];
+        }),
+        tap((productInfo) => {
+          const window = this.winRef.nativeWindow as any;
+          const dispatchEvent = window?.dispatchEvent;
+          if (dispatchEvent) {
+            dispatchEvent(
+              new CustomEvent(CtaEvent.OPF_PRODUCT_AMOUNT_CHANGED, {
+                detail: { productInfo, scriptIdentifiers },
+              })
+            );
+          }
+        })
+      )
+      .subscribe();
+  }
+
+  protected productChangedListener(): void {
+    //// need event listener detecting 'pdp counter changed' event and get its qty value.
```

Comment:
[QUESTION]
Do we plan to release the current empty implementation? Or it's a leftover?